crml/eth-state-oracle/src/lib.rs

```diff
@@ -365,18 +396,85 @@ decl_module! {
 			}
 		}
 
+		/// Deposits a bond which is required to submit challenges
+		/// call requests can't be made if there are no challengers available to challenge them
+		#[weight = 500_000]
+		pub fn deposit_challenger_bond(origin) {
+			let origin = ensure_signed(origin)?;
+
+			// Check account doesn't already have a bond
+			if !Self::challenger_bonds(&origin).is_zero() {
+				// Account already has CPay bonded
+				return Err(Error::<T>::AlreadyBonded.into())
+			};
+			// TODO check if we need max challengers, separate to max relayers
+
+			// check user has the requisite funds to make this bid
+			let fee_currency = T::MultiCurrency::fee_currency();
+			let challenger_bond_amount = T::ChallengerBondAmount::get();
+			// Calculate total bond for a challenger, this is the individual bond amount multiplied by the max relayer responses
+			let total_challenger_bond: Balance = challenger_bond_amount.saturating_mul(Self::max_relayer_responses().into());
+			if let Some(balance_after_bond) = T::MultiCurrency::free_balance(&origin, fee_currency).checked_sub(total_challenger_bond) {
+				// TODO: review behaviour with 3.0 upgrade: https://github.com/cennznet/cennznet/issues/414
+				// - `amount` is unused
+				// - if there are multiple locks on user asset this could return true inaccurately
+				// - `T::MultiCurrency::reserve(origin, asset_id, amount)` should be checking this internally...
+				let _ = T::MultiCurrency::ensure_can_withdraw(&origin, fee_currency, total_challenger_bond, WithdrawReasons::RESERVE, balance_after_bond)?;
+			}
+
+			// try lock funds
+			T::MultiCurrency::reserve(&origin, fee_currency, total_challenger_bond)?;
+			ChallengerBonds::<T>::insert(&origin, total_challenger_bond);
+			Self::deposit_event(Event::<T>::ChallengerBondSet(origin, total_challenger_bond));
+		}
+
+		/// Unbonds an accounts bonded challenger assets
+		#[weight = 500_000]
+		pub fn unbond_challenger_bond(origin) {
+			let origin = ensure_signed(origin)?;
+			// Ensure account has bonded amount
+			let bonded_amount: Balance = Self::challenger_bonds(&origin);
+			ensure!(!bonded_amount.is_zero(), Error::<T>::NotEnoughBonded);
+
+			// Check that there are enough challengers to cover all current requests if this one is removed
+			let current_active_requests = Requests::iter().count();
+			let challenger_count = ChallengerBonds::<T>::iter().count().saturating_sub(1);
+			ensure!(
+				challenger_count.saturating_mul(Self::max_relayer_responses() as usize) >= current_active_requests,
+				Error::<T>::CantUnbondChallenger
+			);
+
+			// Check that there isn't an existing request for the account
+			let responses: Vec<(RequestId, CallResponse<T::AccountId>)> = Responses::<T>::iter().collect();
+			for (_, call_response) in responses {
+				if call_response.relayer == origin {
+					return Err(Error::<T>::CantUnbond.into());
+				}
+			}
```

Comment:
this is a challenger so it should check if there's an inflight challenge rather than response (see `ResponsesChallenged`)